### PR TITLE
 Enabled "keep-alive" and "no-delay" flags for connections to be reused

### PR DESCRIPTION
**Pull request by serkan-ozal**

### CODE REVIEW

**Comment by serkan-ozal** [comment]
As I tested, there is approximately *600 ms* latency to send monitoring data from `eu-west-2` (*London*) to our collector at `us-west-2` (*Oregon) because each time new connection is created and this means that TCP/HTTP handshake every time which causes more RTT. 

By supporting connection reuse with `keep-alive` flag enabled, it is about *160ms* now.

File: src/Reporter.ts [hunk 1]
```diff
@@ -1,15 +1,59 @@
...
 
-    constructor(apiKey: string) {
+    constructor(apiKey: string, u?: url.URL) {
         this.reports = [];
         this.apiKey = apiKey;
+        this.useHttps = (u ? u.protocol : URL.protocol) === 'https:';
+        this.requestOptions = this.createRequestOptions();
+    }
+
+    getConfig(conf1: any, conf2: any): any {
```
[the remark this answers is not in the document]
Comment (serkan-ozal) [reply src/Reporter.ts]: yes, it is left over :+1:

[the remark this answers is not in the document]
Comment (kobalski) [reply src/Reporter.ts]: Merging the pr now and will delete leftover code with another pr.

**Comment by kobalski** [comment]
👍

**Pull request merged**